datalad/interface/list_handles.py

```diff
@@ -12,25 +12,41 @@
 
 __docformat__ = 'restructuredtext'
 
-
-from os import curdir
 from os.path import join as opj, abspath
+
 from .base import Interface
-from datalad.support.param import Parameter
-from datalad.support.constraints import EnsureStr
-from datalad.support.collectionrepo import CollectionRepo
+from ..support.param import Parameter
+from ..support.collectionrepo import CollectionRepo
+from ..support.collectionrepo import CollectionRepoBackend
+
 from appdirs import AppDirs
 
 dirs = AppDirs("datalad", "datalad.org")
 
 
 class ListHandles(Interface):
-    """List all locally installed handles."""
+    """List all locally installed or remote available handles."""
+
+    _params_ = dict(
+        remote=Parameter(
+            args=("-r", "--remote"),
+            action="store_true",
+            doc="""Flag if to list remote handles (not local)"""),
+    )
 
-    def __call__(self):
+    def __call__(self, remote=False):
 
         local_master = CollectionRepo(opj(dirs.user_data_dir,
                                       'localcollection'))
-        for handle in local_master.get_handle_list():
-            print(handle)
+        
+        if remote:
+            for remote_branch in local_master.git_get_remote_branches():
+                if not remote_branch.endswith('/master'): # for now only those
+                    continue
+                for h in CollectionRepoBackend(local_master, branch=remote_branch).get_handles():
+                    remote_name = '/'.join(remote_branch.split('/')[:-1])
+                    print("%s/%s" % (remote_name, h))
+        else:
```

Comment:
it is not '--all' but '--remote' so I felt this way.  Modes of operation are either to list installed handles or available (--remote) not all the handles at once (I don't see when I would need both)